### PR TITLE
feat: mistral inference engine extension

### LINKED ISSUE #2072: feat: inference-mistral-extension
**Problem**

We ❤️ mistral

**Success Criteria**

- similar to openai remote api 
- lets have a mistral remote api

**Additional context**

### LINKED ISSUE #2518: feat: please add cloud-api support for Mistral's 3 models
I tried using the openai and groq api code as a template for creating a Mistral api pathway in Jan...but nothing worked. Whoever set these up could probably add mistral in a few seconds. 
As the misral api is modeled closely after openai

 "https://api.mistral.ai/v1/chat/completions" \

[mistral.json](https://github.com/janhq/jan/files/14771936/mistral.json)
[model.json](https://github.com/janhq/jan/files/14771937/model.json)
[extensions.json](https://github.com/janhq/jan/files/14771938/extensions.json)
**Problem**

**Success Criteria**

**Additional context**
[mistral_tiny_cloud_api.zip](https://github.com/janhq/jan/files/14771960/mistral_tiny_cloud_api.zip)

### PR DESCRIPTION
**Pull request by Inchoker**
## Describe Your Changes

- Add new feature: inference mistral extension

## Fixes Issues

- Closes #2518
- Closes #2072 

## Self Checklist

- [x] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

image

image

### CODE REVIEW

**Comment by louis-jan** [comment]
Blocked by the `registerModels` feature implementation (#2451).

File: core/src/browser/extensions/engines/AIEngine.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export abstract class AIEngine extends BaseExtension {
    * Registers AI Engines
    */
   registerEngine() {
-    EngineManager.instance().register(this)
+    EngineManager.instance().register<AIEngine>(this)
```
Comment (louis-jan) [inline core/src/browser/extensions/engines/AIEngine.ts:32]: Discard irrelevant change

File: extensions/inference-nitro-extension/src/@types/global.d.ts [hunk 2]
```diff
@@ -13,3 +13,14 @@ interface ModelOperationResponse {
   error?: any
   modelFile?: string
 }
+
```
Comment (louis-jan) [inline extensions/inference-nitro-extension/src/@types/global.d.ts:16]: Discard irrelevant change. We will public this as core level.

File: web/screens/ExploreModels/ExploreModelList/index.tsx [hunk 3]
```diff
@@ -30,9 +30,9 @@ const ExploreModelList: React.FC<Props> = ({ models }) => {
         remainingModels.push(m)
       }
     })
-    featuredModels.sort((m1, m2) => m1.name.localeCompare(m2.name))
+    featuredModels.sort((m1, m2) => m1.metadata.size - m2.metadata.size)
     remoteModels.sort((m1, m2) => m1.name.localeCompare(m2.name))
-    localModels.sort((m1, m2) => m1.name.localeCompare(m2.name))
+    localModels.sort((m1, m2) => m1.metadata.size - m2.metadata.size)
     remainingModels.sort((m1, m2) => m1.name.localeCompare(m2.name))
```
Comment (louis-jan) [inline web/screens/ExploreModels/ExploreModelList/index.tsx:36]: Should sort this as well.

File: extensions/inference-mistral-extension/resources/settings.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      "value": "https://api.mistral.ai/v1/chat/completions"
+    }
+  },
+  {
+    "key": "mistral-api-key",
+    "title": "API Key",
+    "description": "The Mistral API uses API keys for authentication. Visit your [API Keys](https://console.mistral.ai/api-keys/) page to retrieve the API key you'll use in your requests.",
+    "controllerType": "input",
+    "controllerProps": {
+      "placeholder": "API Key",
```
Comment (louis-jan) [inline extensions/inference-mistral-extension/resources/settings.json:18]: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

File: extensions/inference-mistral-extension/resources/models.json [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    "format": "api",
+    "settings": {},
+    "parameters": {
+      "max_tokens": 4096,
+      "temperature": 0.7
+    },
+    "metadata": {
+      "author": "Mistral",
+      "tags": [
+        "Classification",
```
Comment (louis-jan) [inline extensions/inference-mistral-extension/resources/models.json:50]: Where did you find it?

File: extensions/inference-mistral-extension/resources/models.json [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    "sources": [
+      {
+        "url": "https://docs.mistral.ai/api/"
+      }
+    ],
+    "id": "mistral-small-latest",
+    "object": "model",
+    "name": "Mistral Small",
+    "version": "1.0",
+    "description": "Mistral...le fromage...et..e'pi",
```
Comment (louis-jan) [inline extensions/inference-mistral-extension/resources/models.json:68]:
```suggestion
    "description": "Mistral Small is the ideal choice for simpe tasks that one can do in builk - like Classification, Customer Support, or Text Generation. It offers excellent performance at an affordable price point.",
```

File: extensions/inference-mistral-extension/resources/models.json [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    "sources": [
+      {
+        "url": "https://docs.mistral.ai/api/"
+      }
+    ],
+    "id": "open-mistral-7b",
+    "object": "model",
+    "name": "Mistral Tiny Cloud Api (7b)",
+    "version": "1.0",
+    "description": "Mistral...le fromage...et..e'pi",
```
Comment (louis-jan) [inline extensions/inference-mistral-extension/resources/models.json:12]:
```suggestion
    "description": "Mistral 7B is the first dense model released by Mistral AI. At the time of the release, it matched the capabilities of models up to 30B parameters.",
```

File: extensions/inference-mistral-extension/resources/models.json [hunk 8]
```diff
@@ -0,0 +1,141 @@
+[
+  {
+    "sources": [
+      {
+        "url": "https://docs.mistral.ai/api/"
+      }
+    ],
+    "id": "open-mistral-7b",
```
Comment (louis-jan) [inline extensions/inference-mistral-extension/resources/models.json:8]: Let's remove mistral tiny

File: extensions/inference-mistral-extension/resources/models.json (hunk 5 above)
Comment (Inchoker) [reply extensions/inference-mistral-extension/resources/models.json:50]: https://docs.mistral.ai/guides/model-selection/
I copy from desc of mistral small

File: extensions/inference-mistral-extension/resources/models.json (hunk 8 above)
Comment (Inchoker) [reply extensions/inference-mistral-extension/resources/models.json]: https://github.com/janhq/jan/blob/34222ca5e6c55a1fba0921eb14538abd50f4db7a/extensions/inference-mistral-extension/resources/models.json#L1

File: extensions/inference-mistral-extension/resources/models.json (hunk 5 above)
Comment (Inchoker) [reply extensions/inference-mistral-extension/resources/models.json]: I removed it https://github.com/janhq/jan/blob/34222ca5e6c55a1fba0921eb14538abd50f4db7a/extensions/inference-mistral-extension/resources/models.json#L1

File: extensions/inference-mistral-extension/resources/settings.json (hunk 4 above)
Comment (Inchoker) [reply extensions/inference-mistral-extension/resources/settings.json]: https://github.com/janhq/jan/blob/40ddb4c3b0c5de48bb11bea5332647d47c83c462/extensions/inference-mistral-extension/resources/settings.json#L18

File: web/screens/ExploreModels/ExploreModelList/index.tsx (hunk 3 above)
Comment (Inchoker) [reply web/screens/ExploreModels/ExploreModelList/index.tsx:36]: [localModels.sort((m1, m2) => m1.metadata.size - m2.metadata.size)](https://github.com/janhq/jan/blob/ea45540d5bd17a239ea5072d10e2332cbf96fafe/web/screens/ExploreModels/ExploreModelList/index.tsx#L33-L35)

Comment (Inchoker) [reply web/screens/ExploreModels/ExploreModelList/index.tsx:36]: https://github.com/janhq/jan/blob/ea45540d5bd17a239ea5072d10e2332cbf96fafe/web/screens/ExploreModels/ExploreModelList/index.tsx#L33-L35

**Review by louis-jan — APPROVED** [review]
LGTM

**Pull request merged**